### PR TITLE
DAOS-3343 control: daos_server network scan shouldn't require a config

### PR DESCRIPTION
**Pull request by jbrosenz**
Decoupled configuration validation from configuration load
	so that daos_server network scan can execute even if the
	configuration file does not specify a valid configuration.

Signed-off-by: Joel Rosenzweig <joel.b.rosenzweig@intel.com>

### CODE REVIEW

**Comment by jbrosenz** [comment]
This version of the proposed solution allows the network scan to proceed even if a provider was not configured in the daos_server.yml.  However, it does still require that the daos_server.yml file exist.  

If we want the network scan to proceed even if no configuration file exists at all, I can make a slight change to allow that.  My initial understanding was that we still want to require the presence of the yml file so I developed the solution for that.  If that's not sufficient, I'll submit a version that corrects that.

**Comment by jbrosenz** [comment]
> I basically like the decoupling. I guess the question is as you said - what's the expected user flow? Do they generate the .yml some other way, then run the network scan and populate it?

The expected flow is that the admin uses the default installation, which includes an empty daos_server.yml, and launches "dmg network scan" to get details about the network devices that are available.  The scan outputs data in a format where an individual network device / provider / numa node can be manually pasted into the daos_server.yml.

File: src/control/server/server.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ const maxIoServers = 2
 func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 	log.Debugf("cfg: %#v", cfg)
 
+	err := cfg.Validate()
+	if err != nil {
+		return err
```
Comment (mjmac) [inline src/control/server/server.go]: Might want to use errors.Wrapf(err, "%s: validation failed", cfg.Path) here just so it's crystal clear to the user that there is a config file they need to fix.

Comment (jbrosenz) [reply src/control/server/server.go]: Done.

**Comment by jbrosenz** [comment]
I took Mike's suggested change and applied that.  I see there is now a CI error.  I'll investigate.

**Pull request merged**